### PR TITLE
Add image name tests around hostnames

### PR DESCRIPTION
**Pull request by stevvooe**
Signed-off-by: Stephen J Day <stephen.day@docker.com>

### CODE REVIEW

File: registry/api/v2/names_test.go [hunk 1]
```diff
@@ -164,6 +164,18 @@ var (
...
 		},
+		{
+			input: "b.gcr.io/test.example.com/my-app", // embedded domain component
+		},
+		// TODO(stevvooe): The following is a punycode domain name that we may
+		// want to allow in the future. Currently, this is not allowed but we
+		// may want to change this in the future. Adding this here as invalid
+		// for the time being.
+		{
+			input:   "xn--n3h.com/myimage", // http://☃.com in punycode
```
Comment (dmp42) [inline registry/api/v2/names_test.go]: Can we go with xn--7o8h.com instead? :)

**Comment by dmp42** [comment]
LGTM

File: registry/api/v2/names_test.go (hunk 1 above)
Comment (stevvooe) [inline registry/api/v2/names_test.go]: Get off my lawn.

Comment (stevvooe) [inline registry/api/v2/names_test.go]: Fixed.

**Comment by RichardScothern** [comment]
LGTM.

**Pull request merged**